### PR TITLE
[ESIMD]Move addc and subb functions out of experimental namespace

### PR DESCRIPTION
**Pull request by fineg74**

### CODE REVIEW

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp [hunk 1]
```diff
@@ -497,77 +497,61 @@ imul(T &rmd, T0 src0, T1 src1) {
...
-__ESIMD_API __ESIMD_NS::simd<uint32_t, N>
-addc(__ESIMD_NS::simd<uint32_t, N> &carry, __ESIMD_NS::simd<uint32_t, N> src0,
-     __ESIMD_NS::simd<uint32_t, N> src1) {
-  std::pair<__ESIMD_DNS::vector_type_t<uint32_t, N>,
-            __ESIMD_DNS::vector_type_t<uint32_t, N>>
-      Result = __esimd_addc<uint32_t, N>(src0.data(), src1.data());
-
-  carry = Result.first;
-  return Result.second;
+__SYCL_DEPRECATED("the function moved out of experimental namespace")
```
Comment (v-klochkov) [inline sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:500]:
```suggestion
__SYCL_DEPRECATED("Please use sycl::ext::intel::esimd::addc(carry, src0, src1);")
```

**Review by v-klochkov — COMMENTED** [review]
Looks good. I have only 1 comment regarding the deprecation message.

File: sycl/include/sycl/ext/intel/esimd/math.hpp [hunk 2]
```diff
@@ -1084,6 +1084,132 @@ ESIMD_INLINE ESIMD_NODEBUG T0 reduce(simd<T1, SZ> v, BinaryOperation op) {
...
+__ESIMD_API uint32_t addc(uint32_t &carry, uint32_t src0, uint32_t src1) {
+  __ESIMD_NS::simd<uint32_t, 1> CarryV = carry;
+  __ESIMD_NS::simd<uint32_t, 1> Src0V = src0;
+  __ESIMD_NS::simd<uint32_t, 1> Src1V = src1;
+  __ESIMD_NS::simd<uint32_t, 1> Res = addc(CarryV, Src0V, Src1V);
+  carry = CarryV[0];
+  return Res[0];
+}
+
+// Performs substraction with borrow of 2 unsigned 32-bit vectors.
```
Comment (turinevgeny) [inline sycl/include/sycl/ext/intel/esimd/math.hpp:1148]:
```suggestion
/// Performs substraction with borrow of 2 unsigned 32-bit vectors.
```

File: sycl/include/sycl/ext/intel/esimd/math.hpp [hunk 3]
```diff
@@ -1084,6 +1084,132 @@ ESIMD_INLINE ESIMD_NODEBUG T0 reduce(simd<T1, SZ> v, BinaryOperation op) {
...
+     __ESIMD_NS::simd<uint32_t, N> src1) {
+  std::pair<__ESIMD_DNS::vector_type_t<uint32_t, N>,
+            __ESIMD_DNS::vector_type_t<uint32_t, N>>
+      Result = __esimd_subb<uint32_t, N>(src0.data(), src1.data());
+
+  borrow = Result.first;
+  return Result.second;
+}
+
+// Performs substraction with borrow of unsigned 32-bit vector and scalar.
```
Comment (turinevgeny) [inline sycl/include/sycl/ext/intel/esimd/math.hpp:1167]:
```suggestion
/// Performs substraction with borrow of unsigned 32-bit vector and scalar.
```

File: sycl/include/sycl/ext/intel/esimd/math.hpp [hunk 4]
```diff
@@ -1084,6 +1084,132 @@ ESIMD_INLINE ESIMD_NODEBUG T0 reduce(simd<T1, SZ> v, BinaryOperation op) {
...
+/// parameter
+template <int N>
+__ESIMD_API __ESIMD_NS::simd<uint32_t, N>
+subb(__ESIMD_NS::simd<uint32_t, N> &borrow, __ESIMD_NS::simd<uint32_t, N> src0,
+     uint32_t src1) {
+  __ESIMD_NS::simd<uint32_t, N> Src1V = src1;
+  return subb(borrow, src0, Src1V);
+}
+
+// Performs substraction with borrow of unsigned 32-bit scalar and vector.
```
Comment (turinevgeny) [inline sycl/include/sycl/ext/intel/esimd/math.hpp:1182]:
```suggestion
/// Performs substraction with borrow of unsigned 32-bit scalar and vector.
```

File: sycl/include/sycl/ext/intel/esimd/math.hpp [hunk 5]
```diff
@@ -1084,6 +1084,132 @@ ESIMD_INLINE ESIMD_NODEBUG T0 reduce(simd<T1, SZ> v, BinaryOperation op) {
...
+/// parameter
+template <int N>
+__ESIMD_API __ESIMD_NS::simd<uint32_t, N>
+subb(__ESIMD_NS::simd<uint32_t, N> &borrow, uint32_t src0,
+     __ESIMD_NS::simd<uint32_t, N> src1) {
+  __ESIMD_NS::simd<uint32_t, N> Src0V = src0;
+  return subb(borrow, Src0V, src1);
+}
+
+// Performs substraction with borrow of 2 unsigned 32-bit scalars.
```
Comment (turinevgeny) [inline sycl/include/sycl/ext/intel/esimd/math.hpp:1197]:
```suggestion
/// Performs substraction with borrow of 2 unsigned 32-bit scalars.
```

**Review by turinevgeny — COMMENTED** [review]
Do you think we need tests for both experimental and non-experimental versions? What is the rule of thumb?

**Review by v-klochkov — APPROVED** [review]

**Comment by turinevgeny** [comment]
> > Do you think we need tests for both experimental and non-experimental versions? What is the rule of thumb?
> 
> I updated the tests only for non-experimental version only to avoid producing deprecation messages while building the tests and save some work later when the experimental API will be actually deleted

Ok, thanks.

**Review by turinevgeny — APPROVED** [review]

**Pull request merged**